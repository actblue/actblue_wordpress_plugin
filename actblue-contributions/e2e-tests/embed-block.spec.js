import {
	clickBlockToolbarButton,
	clickButton,
	createNewPost,
	enablePageDialogAccept,
	getEditedPostContent,
	insertBlock,
} from "@wordpress/e2e-test-utils";

import { selectBlockByName } from "./helpers";

jest.setTimeout(30000);

describe("Wrapper block", () => {
	beforeAll(async () => {
		await enablePageDialogAccept();
	});
	beforeEach(async () => {
		await createNewPost();
	});

	// Tests can be added here by using the it() function
	it("Embed block is available", async () => {
		await insertBlock("ActBlue Embed");
		// Check if block was inserted
		expect(await page.$('[data-type="actblue/embed"]')).not.toBeNull();

		expect(await getEditedPostContent()).toMatchSnapshot();
	});

	it("Add embed url", async () => {
		await insertBlock("ActBlue Embed");
		await selectBlockByName("actblue/embed");

		await page.$eval(
			'input[aria-label="ActBlue Embed URL"]',
			(el) =>
				(el.value = "https://secure.actblue.com/donate/actblue-1-embed")
		);

		const [embedBtn] = await page.$x(
			'//div[@data-type="actblue/embed"]//button[contains(text(), "Embed")]'
		);
		await embedBtn.click();

		expect(await getEditedPostContent()).toMatchSnapshot();
	});
});
