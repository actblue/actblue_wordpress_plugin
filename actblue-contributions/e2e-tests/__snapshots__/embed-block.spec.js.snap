// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper block Add embed url 1`] = `"<!-- wp:actblue/embed /-->"`;

exports[`Wrapper block Embed block is available 1`] = `"<!-- wp:actblue/embed /-->"`;
